homeassistant/components/xbox/binary_sensor.py

```diff
@@ -0,0 +1,143 @@
+from homeassistant.core import callback
+from homeassistant.helpers.entity_registry import (
+    async_get_registry as async_get_entity_registry,
+)
+from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import PresenceData, XboxUpdateCoordinator
+from .const import DOMAIN
+
+PRESENCE_ATTRIBUTES = ["online", "in_party", "in_game", "in_multiplayer"]
+
+
+async def async_setup_entry(hass: HomeAssistantType, config_entry, async_add_entities):
+    """Set up Xbox Live friends."""
+    coordinator: XboxUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id][
+        "coordinator"
+    ]
+
+    update_friends = partial(async_update_friends, coordinator, {}, async_add_entities)
+
+    unsub = coordinator.async_add_listener(update_friends)
+    hass.data[DOMAIN][config_entry.entry_id]["binary_sensor_unsub"] = unsub
+    update_friends()
+
+
+class XboxBinarySensorEntity(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Xbox presence state."""
+
+    def __init__(self, coordinator: XboxUpdateCoordinator, xuid: str, attribute: str):
+        """Initialize Xbox binary sensor."""
+        super().__init__(coordinator)
+        self.xuid = xuid
+        self.attribute = attribute
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique, Home Assistant friendly identifier for this entity."""
+        return f"{self.xuid}_{self.attribute}"
+
+    @property
+    def data(self) -> PresenceData:
+        """Return coordinator data for this console."""
+        return self.coordinator.data.presence[self.xuid]
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        if self.attribute == "online":
+            return self.data.gamertag
+
+        attr_name = " ".join([part.title() for part in self.attribute.split("_")])
+        return f"{self.data.gamertag} {attr_name}"
+
+    @property
+    def entity_picture(self) -> str:
+        """Return the gamer pic."""
+        return self.data.display_pic.replace("&mode=Padding", "")
+
+    @property
+    def is_on(self) -> bool:
+        """Return the status of the requested attribute."""
+        if not self.coordinator.last_update_success:
+            return False
+        return getattr(self.data, self.attribute)
+
+    @property
+    def device_state_attributes(self) -> Optional[Dict[str, str]]:
+        """Return friend attributes."""
+        if self.attribute != "online":
+            return None
+
+        return {
+            "status": self.data.presence_text,
+            "gamer_score": self.data.gamer_score,
+            "account_tier": self.data.account_tier,
+            "tenure": self.data.tenure,
+        }
+
+    @property
+    def entity_registry_enabled_default(self) -> bool:
+        """Return if the entity should be enabled when first added to the entity registry."""
+        return self.attribute == "online"
```

Comment:
Why would you opt to disable entities of users who are offline at the moment of first registering the entity? Seems like a strange choice

Reply:
I think they are creating 4 binaries sensors for each of the four PRESENCE_ATTRIBUTES. Only the "online" one is enabled by default.

Reply:
Still doesn't make sense. If the user is offline all related entities will be disabled and the user would have to go into the registry to enable them just because that user is offline.

Reply:
In their proposed change paragraph. 

> By default, only the "online" binary sensor is enabled. Automatically added, but disabled sensors include status senors for "In a Game", "In a Party", and "In a Multiplayer Session".

Online is not a status in this context, its an "attribute", that's either on or off, which forms the binary sensor.

Reply:
I'm still not sure this is the way to handle it. Probably provide config option to enable/disable extra attributes or generate them by default

Reply:
“Attribute” here is referring to an attribute (or property) on the PresenceData dataclass, not the value of that property.